WordPress/src/main/java/org/wordpress/android/ui/accounts/SmartLockHelper.java

```diff
@@ -106,6 +106,12 @@ public void onResult(@NonNull CredentialRequestResult result) {
 
     public void saveCredentialsInSmartLock(@NonNull final String username, @NonNull final String password,
                                            @NonNull final String displayName, @Nullable final Uri profilePicture) {
+        // need username and password fields for Smart Lock
+        // https://github.com/wordpress-mobile/WordPress-Android/issues/5850
+        if (password.isEmpty() || username.isEmpty()) {
```

Comment:
I think we should use `TextUtils.isEmpty()` here, the `@NonNull` annotation doesn't prevent NPEs. Also we should `AppLog` it, just to make sure we won't miss it if this happen recurrently in the future due to bad input.